### PR TITLE
MM-17311 - Flaky TestGroupStore/UpsertMember

### PR DESCRIPTION
**Pull request by ali-farooq0**
#### Summary
`GroupStore.UpsertMember` uses `CreateAt: model.GetMillis()` to update `GroupMember`. When updated in quick successions, MySQL implementation of RowsAffected returns 0, even if the row exists but no actual changes were made, which eventually causes the test to fail.

- Went with the retry approach for now but in the future we should consider looking at https://github.com/go-sql-driver/mysql#clientfoundrows. Started a discussion: https://community.mattermost.com/core/pl/ugqb4pnfjinobnf6wnrff9gyka
#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-17311

### CODE REVIEW

File: store/storetest/group_store.go [hunk 1]
```diff
@@ -581,15 +582,24 @@ func testUpsertMember(t *testing.T, ss store.Store) {
 	require.Zero(t, d2.DeleteAt)
 
 	// Duplicate composite key (GroupId, UserId)
-	_, err = ss.Group().UpsertMember(group.Id, user.Id)
+	// MySQL's implementation of RowsAffected returns 0 if the
```
Comment (lieut-data) [inline store/storetest/group_store.go]: Contrary to most unit-testing guidelines, I /feel/ like a simpler solution is just to `time.Sleep(1 * time.Millisecond)` with a comment ala `// Ensure UpdateAt > CreateAt`.

File: testlib/helper.go [hunk 2]
```diff
@@ -122,6 +123,10 @@ func (h *MainHelper) Close() error {
 		os.RemoveAll(h.testResourcePath)
 	}
 
+	if r := recover(); r != nil {
```
Comment (lieut-data) [inline testlib/helper.go]: Worth pulling this out as a separate change so we can track its impact discretely?

Comment (ali-farooq0) [reply testlib/helper.go]: Created a PR: https://github.com/mattermost/mattermost-server/pull/11726

**Comment by ali-farooq0** [comment]
@lindalumitchell This is another one of those that doesn't anything userface, strictly a unit-test update. Good to merge?

**Comment by lindalumitchell** [comment]
@ali-farooq0 absolutely; label removed. Thanks!

**Pull request merged**